azure/services/virtualmachines/spec.go

```diff
@@ -291,6 +293,48 @@ func (s *VMSpec) generateSecurityProfile() (*compute.SecurityProfile, error) {
 	}, nil
 }
 
+func (s *VMSpec) generateDiagnosticsProfile() (*compute.DiagnosticsProfile, error) {
+	if s.DiagnosticsProfile.Boot == nil {
+		return &compute.DiagnosticsProfile{
+			BootDiagnostics: &compute.BootDiagnostics{
+				Enabled: to.BoolPtr(true),
+			},
+		}, nil
+	}
+
+	switch s.DiagnosticsProfile.Boot.StorageAccountType {
+	case infrav1.DisabledDiagnosticsStorage:
+		return &compute.DiagnosticsProfile{
+			BootDiagnostics: &compute.BootDiagnostics{
+				Enabled: to.BoolPtr(false),
+			},
+		}, nil
+	case infrav1.ManagedDiagnosticsStorage:
+		return &compute.DiagnosticsProfile{
+			BootDiagnostics: &compute.BootDiagnostics{
+				Enabled: to.BoolPtr(true),
+			},
+		}, nil
+	case infrav1.UserManagedDiagnosticsStorage:
+		if s.DiagnosticsProfile.Boot.UserManaged == nil {
+			return nil, azure.WithTerminalError(fmt.Errorf("`storageURI` must be specified if `storageAccountType` is \"%s\"", infrav1.UserManagedDiagnosticsStorage))
```

Comment:
this should be validated earlier in the webhook

Reply:
Moved